### ISSUE TITLE
Support the `in` operator for More Channel Efficiency

### ISSUE THREAD

**Issue by Marviel** [supabase/realtime#401]
# Feature request

## TL;DR
When fetching a batch of entities by id, the `in` operator is essential for doing so in a performant manner.

Similarly, the `in` operator should be available for realtime subscriptions, in order to save on bandwidth and [realtime subscription rate limits](https://supabase.com/docs/guides/realtime/rate-limits).

Without this feature, it is very hard to scale up any application which uses supabase realtime in a serious capacity, due to the rate limits.

## Detailed Explanation of Why This is Needed

Imagine a **Message Thread** (i.e. Slack Channel, Discord Channel, etc) where users can edit the **Messages** in the **Thread** .

Imagine additionally, there is a a **Notification Menu**, which shows you recent **Notifications** in the application. These Notifications come in many flavors. One flavor of Notification is a **Message Notification**, which contains truncated Messages from various Message Threads.

If a User edits a Message, you would like BOTH the Message displayed in the Notification Menu, AND the Message displayed in the Message Thread, to be updated with the most recent text of that message.

The question is -- how does one subscribe to the change feed for an individual message? There are two possibilities here.
The first possibility, is that the Message Thread component and the Message Notification component both must subscribe to the messages they contain -- i.e. the queries `message:thread_id=THE_THREAD_ID`, and `message:message_notification_id=THE_MESSAGE_NOTIFICATION_ID`.

However this can raise problems.

Firstly,  it's not always _possible_ for the parent component to write a simple supabase query which would properly subscribe. The above subscription we use as an example: 

`message:message_notification_id=THE_MESSAGE_NOTIFICATION_ID`

Is _not actually possible_ to construct in our actual schema, because:
1.  the connection between these objects is mediated by a secondary object called `notification`, and
2. even if it wasn't, the relationship between THOSE two entities is a foreign key in the opposite direction that one would need to use. **_(This is probably another feature request -- reverse foreign key subscriptions)_**

There are other cases where the current single-id-based fetching for objects in collections doesn't make sense -- what if you want to fetch all things where a jsonb field matches a certain id? Of course it's _**nice**_ when you've designed your schema such that everything has an actual direct foreign key to everything it references -- but _**in practice**_, this doesn't always happen. Allowing for fetching by a batch of ids can cover for cases where the user's schema falls short of the ideal.

Secondly, to reference the previous example -- _**even if**_ it's possible to make the subscriptions that were previously mentioned -- the frontend Developers of the NotificationList and MessageList UI components must _remember_ to susbscribe to the change feed for these messages. At [Revaly](https://revaly.com), we're designing an SDK around supabase, and we believe it would ideally be the case that _whenever_ a Message object was requested by _ANY_ UI component, the SDK would be smart enough to automatically subscribe to future changes on that entity -- unless the user has explicitly asked not to.

All this is to say -- without the ability to batch `in` realtime queries, as a "backup" for when schemas are not well designed, or as simply a mechanism to reduce developer cognitive load and default to "responsiveness" -- it becomes much harder to make a snappy user experience

**Comment by chasers** [comment]
Thanks for the detail here @Marviel ... agree we need to add the `in` operator for sure.

**Comment by w3b6x9** [comment]
We have merged this functionality: https://github.com/supabase/realtime/pull/467. Just waiting to release it now.

**Comment by w3b6x9** [comment]
@Marviel this is up on prod now. Make sure to use the format `a=in.(1,2,3)` and it should work.

**Comment by Marviel** [comment]
@w3b6x9 awesome work, very excited for this.
